src/main/java/org/broadinstitute/hellbender/tools/spark/sv/Kmer.java

```diff
@@ -0,0 +1,177 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import java.io.Serializable;
+
+/**
+ * An immutable Kmer.
+ * K must be between 1 and 63 (but it's silly to use this class for K < 33).
+ * Canonicalization is unimplemented for even K.
+ *
+ * Created by tsharpe on 12/10/15.
+ */
+public final class Kmer implements Comparable<Kmer>, Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    // these are treated as K-bit unsigned integers
+    private final long v1; // most significant K bits
+    private final long v2; // least significant K bits
+
+    /**
+     *  Makes an empty Kmer.  If you call toString on it, it'll look like poly-A.
+     */
+    public Kmer( final int kSize )
+    {
+        if ( kSize < 1 || kSize > 63 ) throw new IllegalArgumentException("K must be between 1 and 63.");
+        v1 = v2 = 0;
+    }
+
+    /**
+     * Add a new trailing base to a Kmer, discarding the leading base.
+     * E.g., if kkk.toString(5) is "ACTGA", then kkk.successor(1,5).toString(5) is "CTGAC".
+     * @param base must be 0, 1, 2, or 3, corresponding to A, C, G, or T.
+     */
+    public Kmer successor( final long base, final int kSize )
```

Comment:
It would also be reasonable to use the byte representation of bases (since we use that in most places) .  If you find adding an enum ends up with just an extra layer of translation then you could fall back to that.  

What we want to avoid is adding a bunch of magic numbers that need to be remembered.